core/shared/src/main/scala/zio/Cause.scala

```diff
@@ -83,7 +83,7 @@ sealed trait Cause[+E] extends Product with Serializable { self =>
       case _                 => false
     }
 
-  final def map[E1](f: E => E1): Cause[E1] = self match {
+  final def map[E1 <: Nothing](f: E => E1): Cause[E1] = self match {
```

Comment:
This type bound is not correct (there are no subtypes of `Nothing`). I'd delete it.